src/shk.c

```diff
@@ -1004,6 +1004,7 @@ register struct obj *obj;
 
 	while ((curr = obj->cobj) != 0) {
 	    obj_extract_self(curr);
+	    if (curr->oartifact) artifact_exists(curr, ONAME(curr), FALSE);
```

Comment:
`delete_contents()` should not set artifacts as not being created -- polymorphing a container calls this, as does exploding a bag of holding. It would then be possible to put an artifact inside a container, destroy it, and then get the artifact again.

For bonus points, put a single Fluorite Octahedron in a box, then poly the box, then wish for another Octet and watch the game break.